docs/developer-guide.md

```diff
@@ -0,0 +1,18 @@
+# Design
+
+The system is implemented as an k8s operator using the [operator-sdk](https://github.com/operator-framework/operator-sdk)
+but is deployed as a DaemonSet instead of Deployment with [filtering](https://github.com/operator-framework/operator-sdk/blob/master/doc/user/event-filtering.md) only events for the DaemonSet pod node.
+
+There are two controllers one for [Node](https://godoc.org/k8s.io/api/core/v1#Node)
+and the other for the CRD NodeNeworkState.
+
+## Components
+
+- [Node Controller](developer-guide-state.md) - developer guide and design of `NodeNetworkState` handler.
```

Comment:
I think you need to replace here "developer-guide-state.md" -> "developer-guide-node.md"

Reply:
also the `NodeNetworkState` is fine on both lines?

Reply:
Good catch! done.